physlr/physlr.py

```diff
@@ -1517,6 +1517,45 @@ def determine_molecules_partition_split_merge(g, u):
                     bi_connected_component, strategy=0)
                 ]
 
+    @staticmethod
+    def determine_molecules_extensive(g, u):
+        """
+        Assign the neighbours of this vertex to molecules
+        by Applying a queue of different algorithms on top of each other.
+        """
+        from collections import deque
+
+        communities = [g[u].keys()]
+        communities_final = []
+
+        alg_list = deque(["bc", "k3", "bc", "cos", "bc", "sqCos", "bc"])
+        while alg_list:
+            communities_final.clear()
+            algoritm = alg_list.popleft()
+            if algoritm == "bc":
+                for component in communities:
+                    communities_final.extend(
+                        Physlr.detect_communities_biconnected_components(g, component))
+            elif algoritm == "k3":
+                for component in communities:
+                    communities_final.extend(
+                        Physlr.detect_communities_k_clique(g, component))
+            elif algoritm == "cos":
+                for component in communities:
+                    communities_final.extend(
+                        Physlr.detect_communities_cosine_of_squared(
+                            g, component, squaring=False, threshold=0.4))
+            elif algoritm == "sqCos":
+                for component in communities:
+                    communities_final.extend(
+                        Physlr.detect_communities_cosine_of_squared(g, component))
+            elif algoritm == "louvain":
+                for component in communities:
+                    communities_final.extend(
+                        Physlr.detect_communities_louvain(g, component))
+            communities = communities_final.copy()
```

Comment:
```suggestion
            communities = communities_final
```
What's the purpose of the `copy` here?

Reply:
when I set `communities = communities_final` it seems that they become tied to each other, and int the next iteration when I do `communities_final.clear()` also `communities` become cleared and the function do not work as expected.

Reply:
So if its fine, the PR is ready for another review.

Reply:
In that case use `communities_final = []` rather than `communities_final.clear()` and remove the `.copy()` here.

Reply:
Thank Shaun